src/main.ts

```diff
@@ -0,0 +1,258 @@
+ * that will be linted.
+ */
+export async function main({
+  argv,
+  stdout,
+  stderr,
+  config: { cachedRepositoriesDirectoryPath, defaultProjectNames },
+}: {
+  argv: string[];
+  stdout: SimpleWriteStream;
+  stderr: SimpleWriteStream;
+  config: {
+    cachedRepositoriesDirectoryPath: string;
+    defaultProjectNames: string[];
+  };
+}) {
+  const outputLogger = new OutputLogger({ stdout, stderr });
+  const workingDirectoryPath = process.cwd();
+
+  const inputs = await parseInputs({ argv, outputLogger, defaultProjectNames });
+  /* istanbul ignore next: At the moment, there is no real way that Yargs could fail */
+  if (!inputs) {
+    // Even if `process` changes, it's okay.
+    // eslint-disable-next-line require-atomic-updates
+    process.exitCode = 1;
+    return;
+  }
+  const { templateRepositoryName, projectReferences } = inputs;
+
+  const template = await establishMetaMaskRepository({
+    repositoryReference: templateRepositoryName,
+    workingDirectoryPath,
+    cachedRepositoriesDirectoryPath,
+    outputLogger,
+  });
+
+  await lintProjects({
+    projectReferences,
+    template,
+    workingDirectoryPath,
+    cachedRepositoriesDirectoryPath,
+    outputLogger,
+  });
+}
+
+/**
+ * Parses command-line arguments and establishes parameters for the remainder of
+ * this tool.
+ *
+ * @param args - The arguments to this function.
+ * @param args.argv - The name of this executable and its arguments (as obtained
+ * via `process.argv`).
+ * @param args.outputLogger - Writable streams for output messages.
+ * @param args.defaultProjectNames - The set of MetaMask repositories
+ * that will be linted.
+ * @returns The parsed command-line arguments if parsing was successful, or null
+ * otherwise.
+ */
+async function parseInputs({
+  argv,
+  outputLogger,
+  defaultProjectNames,
+}: {
+  argv: string[];
+  outputLogger: OutputLogger;
+  defaultProjectNames: string[];
+}): Promise<Inputs | null> {
+  const args = await parseCommandLineArguments({ argv, outputLogger });
+
+  if (args) {
+    const projectReferences = args._.length > 0 ? args._ : defaultProjectNames;
+    return {
+      templateRepositoryName: DEFAULT_TEMPLATE_REPOSITORY_NAME,
+      projectReferences,
+    };
+  }
+
+  /* istanbul ignore next: At the moment, there is no real way that Yargs could fail */
+  return null;
+}
+
+/**
+ * Parses the command line using `yargs`.
+ *
+ * @param args - The arguments to this function.
+ * @param args.argv - The name of this executable and its arguments (as obtained
+ * via `process.argv`).
+ * @param args.outputLogger - Writable streams for output messages.
+ */
+async function parseCommandLineArguments({
+  argv,
+  outputLogger,
+}: {
+  argv: string[];
+  outputLogger: OutputLogger;
+}) {
+  let yargsFailure: { message: string; error: Error } | null = null;
+  const onFail = (message: string, error: Error) => {
+    /* istanbul ignore next: Not sure how to produce this */
```

Comment:
I think we'd need to actually _throw an error_ in a command to reproduce this. Assuming that's correct, it makes sense to leave this uncovered since it should be unreachable

Reply:
Ah... okay. Makes sense!